metabase/metabase
# Fix edit dashboard calls more APIs than needed

Closes https://github.com/metabase/metabase/issues/31721

### Description

Previously, when we modify a single dashboard attribute e.g. name, description, auto-apply filters, we'd call `saveDashboardAndCards` and this function calls `PUT /api/dashboard/:id` 2 times plus `PUT /api/dashboard/:id/cards` which I think is unnecessary. I fix it so that we call a new function `saveDashboard` and don't call save dashboard cards because we only change the dashboard attribute.

### How to verify

When changing single dashboard attributes e.g.
- name
- description
- auto-apply filters
- cache duration

There should be only one API call `PUT /api/dashboard/:id`

When changing dashboard via edit mode there should be only 
- 1 `PUT /api/dashboard/:id`
- and 1 `PUT /api/dashboard/:id/cards` 

### Checklist
